CMakeLists.txt

```diff
@@ -14,45 +14,43 @@
 #                                                                         #
 ###########################################################################
 
+# Mandatory call to project
+project(opm-material CXX)
+
 cmake_minimum_required (VERSION 2.8)
 
-set( OPM_COMMON_ROOT "" CACHE PATH "Root directory containing OPM related cmake modules")
 option(SIBLING_SEARCH "Search for other modules in sibling directories?" ON)
 
-if(NOT OPM_COMMON_ROOT)
-   find_package(opm-common QUIET)
+if(SIBLING_SEARCH AND NOT opm-common_DIR)
+  # guess the sibling dir
+  get_filename_component(_leaf_dir_name ${PROJECT_BINARY_DIR} NAME)
+  get_filename_component(_parent_full_dir ${PROJECT_BINARY_DIR} DIRECTORY)
+  get_filename_component(_parent_dir_name ${_parent_full_dir} NAME)
+  #Try if <module-name>/<build-dir> is used
+  get_filename_component(_modules_dir ${_parent_full_dir} DIRECTORY)
+  if(IS_DIRECTORY ${_modules_dir}/opm-common/${_leaf_dir_name})
+    set(opm-common_DIR ${_modules_dir}/opm-common/${_leaf_dir_name})
+  else()
+    string(REPLACE ${PROJECT_NAME} opm-common _opm_common_leaf ${_leaf_dir_name})
+    if(NOT _leaf_dir_name STREQUAL _opm_common_leaf
+        AND IS_DIRECTORY ${_parent_full_dir}/${_opm_common_leaf})
+      # We are using build directories named <prefix><module-name><postfix>
+      set(opm-common_DIR ${_parent_full_dir}/${_opm_common_leaf})
+    elseif(IS_DIRECTORY ${_parent_full_dir}/opm-common)
+      # All modules are in a common build dir
+      set(opm-common_DIR "${_parent_full_dir}/opm-common}")
+    endif()
+  endif()
+else()
+  if(NOT IS_DIRECTORY ${opm-common_DIR})
+    message(WARNING "Value ${opm-common_DIR} passed to variable"
+      " opm-common_DIR is not a directory")
+  endif()
 endif()
 
-if (opm-common_FOUND)
```

Comment:
I just wondered: if opm-common cannot be found now, what error message is displayed to the user? I kinda liked the message deleted below -- of course it had to be removed since it was no longer accurate, but I liked the hand-holding going on there.